### DISCUSSION TITLE
Modal option to not append to document.body?

### DISCUSSION THREAD

**Discussion by alexbot89** [twbs/bootstrap#36859] [Q&A]
A change from 5.1.3 to 5.2.0 for modal checks if the document body contains the modal element, however if the modal element originates from within a shadow-dom `document.body.contains` will not work.

https://github.com/twbs/bootstrap/commit/92e664c921ad6b39d9addd7d991a88f3ea6ea7bc

The bootstrap css is loaded within the shadow-dom, so when the modal is append to document.body, it loses the css styling because the bootstrap css is encapsulated in the shadow-dom not in document scope, this causes issues where the modal won't show up because of missing css or other odd behavior.

Is there some option in the Modal class to ignore appending to document.body, or specify another root?

**Comment by dirkoppelt** [comment]
I know BS does not support Web Components yet, but it should not make it even more difficult to use it with WC and shadow dom. Many projects use BS within shadow dom, it works with some adjustments, but changes like this destroy many existing implementations and should be fixed!
